ca/ca.go

```diff
@@ -646,16 +628,71 @@ func (ca *certificateAuthorityImpl) issuePrecertificateInner(ctx context.Context
 	return certDER, &certProfileWithID{certProfile.name, certProfile.hash, nil}, nil
 }
 
-// verifyTBSCertificateDeterminism verifies that x509.CreateCertificate signing
+// verifyTBSCertIsDeterministic verifies that x509.CreateCertificate signing
 // operation is deterministic and produced identical DER bytes between the given
-// lintCert and leafCert. If this fails it's mississuance, but it's better to
-// know about the problem sooner than later.
-func verifyTBSCertificateDeterminism(lintCert *x509.Certificate, leafCert *x509.Certificate) error {
-	if core.IsAnyNilOrZero(lintCert, leafCert) {
-		return fmt.Errorf("lintCertBytes and certBytes were nil")
-	}
-	if !bytes.Equal(lintCert.RawTBSCertificate, leafCert.RawTBSCertificate) {
-		return fmt.Errorf("mismatch between lintCert and leafCert RawTBSCertificate DER bytes: \"%x\" != \"%x\"", lintCert.RawTBSCertificate, leafCert.RawTBSCertificate)
+// lint certificate and leaf certificate. If the DER byte equality check fails
+// it's mississuance, but it's better to know about the problem sooner than
+// later. The caller is responsible for passing the appropriate valid
+// certificate bytes in the correct position.
+func tbsCertIsDeterministic(lintCertBytes []byte, leafCertBytes []byte) error {
+	if core.IsAnyNilOrZero(lintCertBytes, leafCertBytes) {
+		return fmt.Errorf("lintCertBytes of leafCertBytes were nil")
+	}
+
+	// parseTBSCert is a partial copy of //crypto/x509/parser.go to extract the
+	// RawTBSCertificate field from given DER bytes. It the RawTBSCertificate
+	// field bytes or an error if the given bytes cannot be parsed. This is far
+	// more performant than parsing the entire *Certificate structure with
+	// x509.ParseCertificate().
+	parseTBSCert := func(inputDERBytes *[]byte) ([]byte, error) {
+		if inputDERBytes == nil {
+			return nil, errors.New("inputDERBytes was nil")
+		}
+
+		input := cryptobyte.String(*inputDERBytes)
+		// we read the SEQUENCE including length and tag bytes so that
+		// we can populate Certificate.Raw, before unwrapping the
+		// SEQUENCE so it can be operated on
+		if !input.ReadASN1Element(&input, cryptobyte_asn1.SEQUENCE) {
+			return nil, errors.New("x509: malformed certificate")
+		}
```

Comment:
The way you're using ReadASN1Element followed by ReadASN1 is redundant. It's useful in the original x509 parseCertificate source, because they're _saving_ the result of the ReadASN1Element call. But we're throwing it away, so there's no need to do this extra step that produces an intermediate value with tag and length bytes intact.

```suggestion
```